src/module_3/push_exploration.md

```diff
@@ -0,0 +1,5364 @@
+      <td>active_snoozed</td>
+      <td>0.035236</td>
+    </tr>
+    <tr>
+      <th>7</th>
+      <td>days_since_purchase_variant_id</td>
+      <td>-0.034834</td>
+    </tr>
+    <tr>
+      <th>53</th>
+      <td>product_householdsundries</td>
+      <td>0.024792</td>
+    </tr>
+    <tr>
+      <th>28</th>
+      <td>product_bodyskincare</td>
+      <td>0.024669</td>
+    </tr>
+    <tr>
+      <th>55</th>
+      <td>product_juicesquash</td>
+      <td>0.022988</td>
+    </tr>
+    <tr>
+      <th>18</th>
+      <td>hour</td>
+      <td>0.020846</td>
+    </tr>
+    <tr>
+      <th>45</th>
+      <td>product_feedingweaning</td>
+      <td>-0.013937</td>
+    </tr>
+    <tr>
+      <th>6</th>
+      <td>count_pets</td>
+      <td>0.012153</td>
+    </tr>
+    <tr>
+      <th>1</th>
+      <td>order_id</td>
+      <td>0.011454</td>
+    </tr>
+    <tr>
+      <th>2</th>
+      <td>user_id</td>
+      <td>-0.010594</td>
+    </tr>
+    <tr>
+      <th>20</th>
+      <td>product_babyfood12months</td>
+      <td>0.010576</td>
+    </tr>
+    <tr>
+      <th>17</th>
+      <td>day_of_month</td>
+      <td>-0.009731</td>
+    </tr>
+    <tr>
+      <th>68</th>
+      <td>product_shavinggrooming</td>
+      <td>-0.002280</td>
+    </tr>
+    <tr>
+      <th>71</th>
+      <td>product_superfoodssupplements</td>
+      <td>-0.001916</td>
+    </tr>
+  </tbody>
+</table>
+</div>
+
+
+
+Model behaviour it's pretty similar for P = 0.5 in the confusion matrix. Let's try to plot the curves, specially the precision-recall again
+
+
+```python
+# Calculate probabilities for the positive class
+y_val_probs = model.predict_proba(X_val)[:, 1]
+
+# ROC Curve
+fpr_val, tpr_val, _ = roc_curve(y_val, y_val_probs)
+roc_auc_val = auc(fpr_val, tpr_val)  # Calculate AUC
+
+plt.figure(figsize=(10,5))
+
+plt.subplot(1, 2, 1)
+plt.plot(fpr_val, tpr_val, label=f'ROC (AUC = {roc_auc_val:.2f})')
+plt.plot([0, 1], [0, 1], color='navy', linestyle='--')
+plt.xlim([0.0, 1.0])
+plt.ylim([0.0, 1.05])
+plt.xlabel('False Positive Rate')
+plt.ylabel('True Positive Rate')
+plt.title('ROC')
+plt.legend(loc="lower right")
+
+
+# Precision-Recall Curve
+precision_val, recall_val, _ = precision_recall_curve(y_val, y_val_probs)
+
+plt.subplot(1, 2, 2)
+plt.plot(recall_val, precision_val)
+plt.xlim([0.0, 1.0])
+plt.ylim([0.0, 1.05])
+plt.xlabel('Recall')
+plt.ylabel('Precision')
+plt.title('Precision-Recall Curve')
+
+plt.tight_layout()
+plt.show()
+```
+
+
+    
+![png](push_exploration_files/push_exploration_89_0.png)
+    
+
+
+ROC is similar but we see a different pattern in the precision vs recall behaviour. Let's explote them using different threshold for maximising precison while we get a sufficient recall to appreciate positive values. We see that for same recall as we selected before (0.32 approx) we obtain similar or even worse results. However, if we select a little fewer recall, we obtain significant increases in precision vs previous model. Let's use 0.3, 0.2, and 0.15.
```

Comment:
I think you should try to have a consistent metric for all the models you are comparing against each other. This idea of looking at the shape of the PR curve and comparing different regions is not a very good one, since it's not systematic and you can choose which parts to compare each time.